src/mail/view/ConversationViewer.ts

```diff
@@ -0,0 +1,342 @@
+				// there's a chance that item are not in sync with dom but it's very unlikely, this is the next frame after the last render we used the items
+				// and viewModel is finished.
+				const itemIndex = items.findIndex((e) => isSameId(e.entryId, conversationId))
+				// Don't scroll if it's already the first (or if we didn't find it but that would be weird)
+				if (itemIndex > 0) {
+					// There is a small chance that between finding the index and
+					const top = (containerDom.childNodes[itemIndex] as HTMLElement).offsetTop
+					// 46 for the floating header
+					containerDom.scrollTo({ top: top - 46 })
+				}
+			})
+		}
+	}
+
+	private scrollUp(): void {
+		if (this.containerDom) {
+			this.containerDom.scrollBy({ top: -this.containerDom.clientHeight * SCROLL_FACTOR, behavior: "smooth" })
+		}
+	}
+
+	private scrollDown(): void {
+		if (this.containerDom) {
+			this.containerDom.scrollBy({ top: this.containerDom.clientHeight * SCROLL_FACTOR, behavior: "smooth" })
+		}
+	}
+
+	private scrollToTop(): void {
+		if (this.containerDom) {
+			this.containerDom.scrollTo({ top: 0, behavior: "smooth" })
+		}
+	}
+
+	private scrollToBottom(): void {
+		if (this.containerDom) {
+			this.containerDom?.scrollTo({ top: this.containerDom.scrollHeight - this.containerDom.offsetHeight, behavior: "smooth" })
+		}
+	}
+}
+
+type SubjectVisiblity = "above" | "below" | "visible"
+
+interface ObservableSubjectAttrs {
+	cb: (visibility: SubjectVisiblity) => unknown
+	subject: string
+}
+
+export class ObservableSubject implements Component<ObservableSubjectAttrs> {
+	lastAttrs: ObservableSubjectAttrs
+
+	observer: IntersectionObserver | null = null
+
+	constructor(vnode: Vnode<ObservableSubjectAttrs>) {
+		this.lastAttrs = vnode.attrs
+	}
+
+	view(vnode: Vnode<ObservableSubjectAttrs>): Children {
+		this.lastAttrs = vnode.attrs
+		return m(
+			".h5.subject.text-break.selectable.b.flex-grow.mt-m",
+			{
+				class: mailViewerMargin(),
+				"aria-label": lang.get("subject_label") + ", " + (this.lastAttrs.subject || ""),
+				style: { marginTop: "12px" },
+				oncreate: (vnode) => {
+					this.observer = new IntersectionObserver(
+						(entries) => {
+							const [entry] = entries
+							const visibility = entry.isIntersecting
+								? "visible"
+								: entry.boundingClientRect.bottom < assertNotNull(entry.rootBounds).top
+								? "above"
+								: "below"
+							this.lastAttrs.cb(visibility)
+						},
+						{ root: vnode.dom.parentElement },
+					)
+					this.observer.observe(vnode.dom)
+				},
+				onremove: (vnode) => {
+					this.observer?.unobserve(vnode.dom)
+				},
+			},
+			this.lastAttrs.subject,
+		)
+	}
+}
+
+class DeletedMailView implements Component {
+	view() {
+		return m(
+			".center.pt-s.pb-s.font-weight-600.border-radius-big.mt-m",
+			{
+				class: mailViewerMargin(),
+				style: {
+					border: `1px solid ${theme.list_border}`,
+					color: theme.content_button,
+				},
+			},
+			"Deleted Email",
```

Comment:
This needs at least a fixme, "Send Logs" was untranslated for months

Reply:
I was against adding a FIXME because we don't merge code with FIXME's but I also agree with you.

Reply:
this is not about merging, wouldn't want it on master or in any release, just as a reminder that we need to add the translation key before doing anything serious with the PR